pkg/cmd/join/exec.go

```diff
@@ -92,6 +95,16 @@ func (o *Options) validate() error {
 		return fmt.Errorf("registry should not be empty")
 	}
 
+	// preflight check
+	if err := preflightinterface.RunChecks(
+		[]preflightinterface.Checker{
+			preflight.ApiHostIPCheck{
+				ApiHostIP: o.values.Hub.APIServer,
```

Comment:
is this the value read from kube-public config map?

Reply:
done